### ISSUE TITLE
M4.3: Train explainable logistic-regression ranking baseline

### ISSUE THREAD

**Issue by joshuaokent-spec** [joshuaokent-spec/JobOps#46]
## Goal
Train and evaluate the first learned JobOps job-ranking model using only M4.2 model-ready rows and the fixed temporal/feature contract.

## Model
Start with logistic regression as the transparent supervised baseline.

The model must consume the M4.2 feature schema rather than rebuilding features or labels independently.

## Scope
- load M4.2 `rows.jsonl` + `manifest.json`
- validate dataset fingerprint/schema/builder version before training
- use train rows only for fitting and validation rows only for evaluation
- no random reshuffle of the temporal split
- encode categorical features deterministically
- standardize numeric features where appropriate
- preserve a versioned feature order / preprocessing contract
- report coefficients in original feature-name terms where possible
- calculate probability outputs for the positive ranking target
- persist model metadata separately from candidate/private training data

## Metrics
Report at minimum:
- ROC AUC when both classes are present
- PR AUC / average precision
- log loss
- Brier score
- accuracy at a documented threshold
- precision / recall / F1
- confusion counts
- validation class balance

Do not fabricate metrics when validation contains only one class; report them as unavailable with diagnostics.

## Calibration
The baseline should expose raw logistic probabilities and enough metadata for the later calibration/drift milestone. Do not claim calibration quality from training fit alone.

## Explainability
Persist:
- intercept
- transformed feature names
- coefficients
- categorical vocabulary
- numeric preprocessing parameters
- dataset fingerprint
- training cutoff range
- model version
- evaluation metrics

## Reproducibility
Given the same M4.2 dataset and model configuration, training should be deterministic.

Use a fixed solver/configuration and record all relevant hyperparameters.

## Comparison
Compare learned validation performance with the existing deterministic baseline score where meaningful. Treat this as analysis, not a requirement that the learned model must beat the heuristic on a tiny synthetic dataset.

## Persistence / output
First version may write a private model bundle such as:

```text
models/ranking/logistic-v1/
  model.joblib
  metadata.json
  evaluation.json
```

Model artifacts generated from production candidate data must remain ignored/private.

## Tests
- rejects mismatched/unsupported M4.2 manifest
- trains on train split only
- evaluates validation only
- deterministic repeated fit produces identical coefficients/predictions
- categorical unknown handling is defined
- single-class validation returns diagnostics instead of crashing
- model metadata preserves dataset fingerprint + feature schema
- no raw narrative/PII fields enter preprocessing
- synthetic fixture only

## Acceptance criteria
Given a valid synthetic M4.2 dataset, JobOps can train, persist, reload, and evaluate a deterministic logistic-regression ranking baseline whose lineage points back to the exact dataset fingerprint.

## Portfolio value
Demonstrates supervised ML modeling, temporal holdout evaluation, deterministic preprocessing, probability metrics, model lineage, explainability, and the transition from rule-based ranking to learned ranking without abandoning auditability.
